File: aj-server/src/main/java/com/ajinternational/ajserver/modules/masterdata/model/MasterProduct.java
package com.ajinternational.ajserver.modules.masterdata.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

// 'parentProductId' ve 'subProducts' kaldırıldı.

@Data
@NoArgsConstructor
@Document(collection = "master_products")
public class MasterProduct {

    @Id
    private String id;

    @Indexed
    private String tenantId;

    @Indexed(unique = true) // Bu (kod) aynı kalıyor
    private String code;

    private String name;

    private String description;

    // --- YENİ ALANLAR (SADECE ALT ÜRÜNLER İÇİN) ---
    // Bu alanlar eski modelde zaten vardı, isimleri değişmedi
    private Double targetValue;
    private String measureDefinitionId;
    private boolean active = true;
    private Double wasteRate;

    // --- GÜNCELLENEN ALANLAR ---
    // 1. Ürünün hangi üretim bölümüne bağlı olduğunu tutar
    @Indexed
    private String productionUnitId; // (Yeni) 'parentProductId' yerine geldi

    // 2. 'premiumValue' alanı 'unitPrice' oldu
    private Double unitPrice; // (Değişti) 'premiumValue' yerine geldi


    // --- GEÇİCİ ALANLAR (Frontend'e veri taşımak için) ---
    @Transient
    private String groupName; // (Yeni) Bu ürünün bağlı olduğu Grubun adı

    @Transient
    private String sectionName; // (Yeni) Bu ürünün bağlı olduğu Bölümün adı
}

File: aj-client/src/types/master-product.ts
export type MasterProduct = {
  id: string;
  tenantId: string;
  code: string;
  name: string;
  description?: string;
  
  // Hiyerarşi kaldırıldı, yerine bu geldi
  productionUnitId: string | null; // Bölüm ID'si
  
  // subProducts kaldırıldı.
  
  active: boolean;
  targetValue?: number | null;
  measureDefinitionId?: string | null;
  wasteRate?: number | null;
  
  // Değişen alan
  unitPrice?: number | null; // 'premiumValue' yerine

  // Backend servisinden join'lenerek gelen ekstra alanlar
  groupName?: string;
  sectionName?: string;
};

// Form tipi de güncellendi
export type MasterProductFormValues = {
  id?: string;
  name: string;
  code: string;
  description?: string;
  
  productionUnitId: string | null;
  
  active: boolean;
  targetValue?: number | null;
  measureDefinitionId?: string | null;
  wasteRate?: number | null;
  unitPrice?: number | null;
};